lib/ansible/plugins/action/script.py

```diff
@@ -87,41 +87,47 @@ def run(self, tmp=None, task_vars=None):
             except AnsibleError as e:
                 raise AnsibleActionFail(to_native(e))
 
+            if self._task.check_mode:
+                # check mode is supported if 'creates' or 'removes' are provided
+                # the task has already been skipped if a change would not occur
+                if self._task.args.get('creates') or self._task.args.get('removes'):
+                    result['changed'] = True
+                    return result
+                # If the script doesn't return changed in the result, it defaults to True,
+                # but since the script may override 'changed', just skip instead of guessing.
+                else:
+                    result['changed'] = False
+                    raise AnsibleActionSkip('Check mode is not supported for this task.')
```

Comment:
raise AnsibleActionSkip('Check mode is not supported for this task.', result=result)